File: additional/internal/services/cmd/forward_motion/forward_motion.go
// Пакет для обработки макрокоманды длительного движения по прямой
package forwardmotion

import (
	"additional/internal/services/cmd"
	"additional/internal/services/cmd/macros"
)

type ForwardMotion struct {
	macros macros.Macros
}

func NewForwardMotion(check cmd.ICommand, move cmd.ICommand, refuel cmd.ICommand) *ForwardMotion {
	queue := cmd.NewCommandQueue()
	queue.Add(check)
	queue.Add(move)
	queue.Add(refuel)
	macros := macros.NewMacros(*queue)
	return &ForwardMotion{macros: *macros}
}

func (m *ForwardMotion) Execute() error {
	return m.macros.Execute()
}

File: main/src/server/cmd/forward_motion.py
from queue import Queue

import zope

from server.cmd.macros import MacrosCommand
from server.interfaces.cmd import ICommand


@zope.interface.implementer(ICommand)
class ForwardMotionCommand:
    """Макрокоманда длительного движения по прямой"""

    def __init__(self, check, move, burn):
        self._queue = Queue()
        self._queue.put(check)
        self._queue.put(move)
        self._queue.put(burn)
        self.macros = MacrosCommand(self._queue)

    def execute(self) -> None:
        self.macros.execute()
